# Question 2

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


### Loading dataset onto a dataframe

In [5]:
rows = [
    [2 , 4, 2, 2, 5, 65],
    [3, 3, 4, 3, 6, 70],
    [4, 2, 6, 4, 5, 75],
    [5, 1, 8, 1, 6, 80],
    [6, 0, 10, 5, 4, 85],
    [7, 0, 12, 0, 5, 90]
]
df = pd.DataFrame(rows, columns = ['Hours_Studied', 'Hours_Watching_TV', 'Outdoor_Activity_Time', 'Hours_Listening_to_Music', 'Water_Consumed', 'Test_Score'])
df.reset_index(drop=True)
print(df)

   Hours_Studied  Hours_Watching_TV  Outdoor_Activity_Time  \
0              2                  4                      2   
1              3                  3                      4   
2              4                  2                      6   
3              5                  1                      8   
4              6                  0                     10   
5              7                  0                     12   

   Hours_Listening_to_Music  Water_Consumed  Test_Score  
0                         2               5          65  
1                         3               6          70  
2                         4               5          75  
3                         1               6          80  
4                         5               4          85  
5                         0               5          90  


### Correlation Coefficient Calculation

ρX,Y = cov(X,Y) / σX * σY

In [6]:

#covariance of (X, Y)
#for cov youre basically finding the diff between each variable's mean and sample and then multiply with each other
counter = 0
Hours_Studied_mean = 0
Hours_Watching_TV_mean = 0
Outdoor_Activity_Time_mean = 0
Hours_Listening_to_Music_mean = 0
Water_Consumed_mean = 0
Test_Score_mean = 0

for ind,row in df.iterrows():
  counter = counter + 1
  Hours_Studied_mean = Hours_Studied_mean + row.Hours_Studied
  Hours_Watching_TV_mean = Hours_Watching_TV_mean + row.Hours_Watching_TV
  Outdoor_Activity_Time_mean = Outdoor_Activity_Time_mean + row.Outdoor_Activity_Time
  Hours_Listening_to_Music_mean = Hours_Listening_to_Music_mean + row.Hours_Listening_to_Music
  Water_Consumed_mean = Water_Consumed_mean + row.Water_Consumed
  Test_Score_mean = Test_Score_mean + row.Test_Score

Hours_Studied_mean = Hours_Studied_mean / counter
Hours_Watching_TV_mean = Hours_Watching_TV_mean / counter
Outdoor_Activity_Time_mean = Outdoor_Activity_Time_mean / counter
Hours_Listening_to_Music_mean = Hours_Listening_to_Music_mean / counter
Water_Consumed_mean = Water_Consumed_mean / counter
Test_Score_mean = Test_Score_mean / counter

#Hours watching TV to Test Score
#Outdoor Activity Time to Test Score
#Hours Listening to Music to Test Score
#Water Consumed
HS_TS_cov = 0
TV_TS_cov = 0
OAT_TS_cov = 0
LM_TS_cov = 0
WC_TS_cov = 0

HS_var = 0
TV_var = 0
OAT_var = 0
LM_var = 0
WC_var = 0
TS_var = 0
for ind,row in df.iterrows():
  HS_TS_cov = HS_TS_cov + ((row.Hours_Studied - Hours_Studied_mean)*(row.Test_Score - Test_Score_mean))
  TV_TS_cov = TV_TS_cov + ((row.Hours_Watching_TV - Hours_Watching_TV_mean)*(row.Test_Score - Test_Score_mean))
  OAT_TS_cov = OAT_TS_cov + ((row.Outdoor_Activity_Time - Outdoor_Activity_Time_mean)*(row.Test_Score - Test_Score_mean))
  LM_TS_cov = LM_TS_cov + ((row.Hours_Listening_to_Music - Hours_Listening_to_Music_mean)*(row.Test_Score - Test_Score_mean))
  WC_TS_cov = WC_TS_cov + ((row.Water_Consumed - Water_Consumed_mean)*(row.Test_Score - Test_Score_mean))

  HS_var = HS_var + ((row.Hours_Studied - Hours_Studied_mean)**2)
  TV_var = TV_var + ((row.Hours_Watching_TV - Hours_Watching_TV_mean)**2)
  OAT_var = OAT_var + ((row.Outdoor_Activity_Time - Outdoor_Activity_Time_mean)**2)
  LM_var = LM_var + ((row.Hours_Listening_to_Music - Hours_Listening_to_Music_mean)**2)
  WC_var = WC_var + ((row.Water_Consumed - Water_Consumed_mean)**2)
  TS_var = TS_var + ((row.Test_Score - Test_Score_mean)**2)

HS_TS_cor_coef = HS_TS_cov / np.sqrt(HS_var * TS_var)
TV_TS_cor_coeff = TV_TS_cov / np.sqrt(TV_var * TS_var)
OAT_TS_cor_coeff = OAT_TS_cov / np.sqrt(OAT_var * TS_var)
LM_TS_cor_coeff = LM_TS_cov / np.sqrt(LM_var * TS_var)
WC_TS_cor_coeff = WC_TS_cov / np.sqrt(WC_var * TS_var)

### Based on the provided data, calculate and interpret the correlation coefficient for each variable in relation to the test scores. Identify the variables that show a positive correlation, a negative correlation, and no significant correlation with the test scores.


Explain why there might be a positive correlation between the "Hours Studied" variable and the test scores. Provide a brief discussion on how this information could be valuable for improving academic performance.


Calculate and interpret the correlation coefficient between "Hours Watching TV" and test scores. Explain the implications of this correlation in terms of academic achievement and time management.


Calculate and interpret the correlation coefficient between "Hours Listening to Music" and test scores. Discuss the potential impact of this correlation on concentration and study habits.


Calculate and interpret the correlation coefficient between "Water Consumed" and test scores. Provide a potential explanation for the observed correlation and its relevance to cognitive function.


Calculate and interpret the correlation coefficient between "Outdoor Activity Time" and test scores. Discuss how physical activity might influence academic performance.


You are not allowed to use built-in functions to calculate correlation. Instead, you should perform the calculations manually, showing all steps clearly. Analyze the correlations between each variable and the test scores to draw meaningful insights from the data.

####*Calculate and interpret the correlation coefficient between "Hours Studied" and test scores.*

In [7]:
print("Correlation Coefficient for Hours Studied and Test Scores : ")
print(HS_TS_cor_coef)

Correlation Coefficient for Hours Studied and Test Scores : 
1.0


####*Explain why there might be a positive correlation between the "Hours Studied" variable and the test scores. Provide a brief discussion on how this information could be valuable for improving academic performance.*

The correlation coefficient between Hours Studied and Test Scores is +1, implying theres a high positive correlation. High positive correlation means that as the number of "Hours Studied" increases, the "Test Scores" will also increase, i.e. Hours Studied has. great positive impact on Test Scores. Higher Test Scores mean that the student is performing well in academics. For a student with a poor academic performance - low test score, he/she could increase the number of hours they spend studying which can aid in improving their academic performance according to the correlation.

The correlation between "Hours Studied" and "Test Scores" can guide students in adopting more effective study habits. It encourages them to allocate ample time for focused learning, adopt strategies that enhance retention, and engage in meaningful practice. Furthermore, it emphasizes the concept of incremental progress, where even small increases in study time can lead to noticeable improvements in performance over time.

Hours Studied serves as a guiding principle for students aiming to enhance their performance, indicating that their investment in studying can yield substantial returns in terms of improved test scores and overall academic achievement.

####*Calculate and interpret the correlation coefficient between "Hours Watching TV" and test scores.*

In [8]:
print("Correlation Coefficient for Hours Watching TV and Test Scores : ")
print(TV_TS_cor_coeff)

Correlation Coefficient for Hours Watching TV and Test Scores : 
-0.9819805060619656


####*Explain the implications of this correlation between in terms of academic achievement and time management.*

The correlation coefficient between Hours Studied and Test Scores is alomst -1, implying theres a high negative correlation. High negative correlation means that as the number of "Hours Watching TV" increases, the "Test Scores" will decrease, i.e. Hours Studied has a great negative impact on Test Scores. Higher Test Scores mean that the student is performing well in academics. For a student with a poor academic performance - low test score, he/she could decrease the number of hours they spend watching TV which can aid in improving their academic performance according to the correlation.

For students grappling with poor academic performance, as indicated by low test scores, a constructive strategy would involve reducing the time spent on television consumption. By doing so, they can potentially initiate a positive transformation in their academic standing, as suggested by the negative correlation. This correlation underscores the critical nature of effective time management and the allocation of adequate hours to focused studying for enhancing educational outcomes.

The nearly -1 correlation coefficient between Hours Studied and Test Scores highlights a significant negative correlation, signifying that an increase in television viewing hours is associated with a decrease in test performance. To excel academically, students struggling with low test scores can leverage this insight by minimizing TV time and redirecting that time towards productive study sessions, thereby facilitating a potential improvement in their academic achievements.

####*Calculate and interpret the correlation coefficient between "Hours Listening to Music" and test scores.*

In [10]:
print("Correlation Coefficient for Hours Listening to Music and Test Scores : ")
print(LM_TS_cor_coeff)

Correlation Coefficient for Hours Listening to Music and Test Scores : 
-0.2


#### *Discuss the potential impact of this correlation on concentration and study habits.*

The correlation coefficient of -0.2 between Hours Listening to Music and Test Scores implies a weak negative correlation between these two variables. This means that there is a slight tendency for increased hours of listening to music to be associated with slightly lower test scores, but the relationship is not very strong.

Even though the correlation is weak, it does suggest that there might be some interference caused by excessive music listening during study sessions. Music, particularly with lyrics or complex melodies, can divert attention and cognitive resources away from studying. It might lead to difficulties in focusing on complex or information-intensive material.

The impact of music on concentration can also depend on the nature of the task at hand. While music might be fine for certain tasks that require creativity or less intense focus, it might hinder performance on tasks that demand sustained attention and analytical thinking.

Creating an optimal study environment is crucial for effective learning. Students who are more sensitive to the potential distractions posed by music might benefit from quieter study spaces that allow for deep concentration. On the other hand, those who find music motivating and soothing can strategically integrate it into their study routines.

####*Calculate and interpret the correlation coefficient between "Water Consumed" and test scores.*

In [11]:
print("Correlation Coefficient for Water Consumed and Test Scores : ")
print(WC_TS_cor_coeff)

Correlation Coefficient for Water Consumed and Test Scores : 
-0.3550358012483632


####*Provide a potential explanation for the observed correlation and its relevance to cognitive function.*

The correlation coefficient of -0.355 between Water Consumed and Test Scores suggests a moderate negative correlation between these two variables. This indicates that there is a noteworthy relationship where increased water consumption is associated with somewhat lower test scores.

The timing of water consumption in relation to studying and test-taking can influence its impact. Drinking an appropriate amount of water before and during study sessions and tests is important. If students consume excessive amounts of water right before or during the test, it might lead to distractions due to frequent bathroom breaks or discomfort, thereby affecting their test performance.



####*Calculate and interpret the correlation coefficient between "Outdoor Activity Time" and test scores*

In [12]:
print("Correlation Coefficient for Outdoor Activity Time and Test Scores : ")
print(OAT_TS_cor_coeff)

Correlation Coefficient for Outdoor Activity Time and Test Scores : 
1.0


#### *Discuss how physical activity might influence academic performance.*

The correlation coefficient of 1.0 between Outdoor Activity Time and Test Scores indicates a perfect positive correlation between these two variables. This means that there is a strong and direct relationship: as the amount of time spent in outdoor activities increases, test scores also increase.

Physical activity triggers the release of neurotransmitters like dopamine and endorphins, which promote feelings of happiness, reduce stress, and improve mood. These positive emotional states can contribute to a conducive learning environment.

 Outdoor activities offer an opportunity to disconnect from academic pressures. Exposure to natural settings and fresh air can alleviate anxiety, enabling students to approach studies with a clearer and calmer perspective.

 Regular physical activity contributes to overall health, which in turn supports mental well-being. A healthy body and mind synergize to create an environment conducive to effective learning and academic achievement.

 Success in physical activities, such as sports or outdoor challenges, can boost self-esteem and self-efficacy. These positive feelings can spill over into academic pursuits, encouraging students to approach challenges with confidence.